robottelo/hosts.py

```diff
@@ -1848,8 +1848,30 @@ def cli(self):
     @contextmanager
     def omit_credentials(self):
         self.omitting_credentials = True
+        # if CLI is already created
+        if self._cli._configured:
+            for file in Path('robottelo/cli/').iterdir():
+                if file.suffix == '.py' and not file.name.startswith('_'):
+                    cli_module = importlib.import_module(f'robottelo.cli.{file.stem}')
+                    for name, obj in cli_module.__dict__.items():
+                        try:
+                            if Base in obj.mro():
+                                getattr(self._cli, name).omitting_credentials = True
+                        except AttributeError:
+                            # not everything has an mro method, we don't care about them
+                            pass
         yield
         self.omitting_credentials = False
```

Comment:
This is not necessarily `False`. One can initialize Satellite with `True`. In such scenario, this wouldn't reset the attribute to the correct original version.
Perhaps this contextmanager function could check if the `ommiting_credentials`  is set to `False` first, otherwise, it has no reason to run and could be skipped straight away.